src/core/int128.d

```diff
@@ -453,36 +453,135 @@ pure
 Cent udivmod(Cent c1, Cent c2, out Cent modulus)
 {
     //printf("udiv c1(%llx,%llx) c2(%llx,%llx)\n", c1.lo, c1.hi, c2.lo, c2.hi);
+    // Based on "Unsigned Doubleword Division" in Hacker's Delight
+    import core.bitop;
+
+    // Divides a 128-bit dividend by a 64-bit divisor.
+    // The result must fit in 64 bits.
+    static U udivmod128_64(Cent c1, U c2, out U modulus)
```

Comment:
@ibuclaw  FYI, on x86_64 this function can be implemented using a single instruction.

https://github.com/CyberShadow/ae/blob/88e04cedb550a4224142207852eb350a01a3c0b1/utils/math/longmul.d#L147